include/drivers/gpio/gpio_esp32.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __GPIO_ESP32_H__
+#define __GPIO_ESP32_H__
+
+struct enable_bits {
```

Comment:
I'd prefix these structs with `esp32_` just to be safe.  Also, since this is in a header, it might be a good idea to add some documentation.

This could move to `soc.h`, since it's usable outside `gpio` drivers as well.